safeweb/http.go

```diff
@@ -376,10 +343,38 @@ func (s *Server) serveBrowser(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Security-Policy", s.csp)
 	w.Header().Set("X-Content-Type-Options", "nosniff")
 	w.Header().Set("Referer-Policy", "same-origin")
-	if s.SecureContext {
+	if s.ServeHSTS {
 		w.Header().Set("Strict-Transport-Security", cmp.Or(s.StrictTransportSecurityOptions, DefaultStrictTransportSecurityOptions))
 	}
-	s.csrfProtect(s.BrowserMux).ServeHTTP(w, r)
+
+	// Allow GET, HEAD, and OPTIONS requests to the browser mux without
+	// Sec-Fetch-Site header checks.
+	switch r.Method {
+	case "GET", "HEAD", "OPTIONS":
+		s.BrowserMux.ServeHTTP(w, r)
+		return
+	default:
```

Comment:
FWIW, the `default` isn't needed here (although it is also harmless).

(This is in contrast with a `select` where a default case has a material effect on the fallthrough behaviour)

Reply:
an educational drop-in comment is the best type. TIL. thanks!